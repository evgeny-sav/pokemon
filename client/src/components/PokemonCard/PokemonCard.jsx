import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import Button from 'react-bootstrap/Button';
import capitalize from '../../utils/capitalize';
import './PokemonCard.scss';

const PokemonCard = ({ pokemon }) => {
  const {
    name,
    sprites: { front_default },
  } = pokemon;

  return (
    <div className="PokemonCard">
      <div className="image">
        <img src={front_default} alt="Pokemon's image" />
      </div>
      <div className="cord-body">
        <h5 className="title">{capitalize(name)}</h5>
        <p className="description">
          Some quick example text to build on the card title and make up the
          bulk of the cards content.
        </p>
        <div className="button">
          <Link to={`/pokemons/${pokemon.id}`}>
            <Button variant="outline-info">Open</Button>
          </Link>
        </div>
      </div>
    </div>
  );
};

PokemonCard.propTypes = {
  pokemon: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    sprites: PropTypes.objectOf(PropTypes.string),
  }),
};

export default PokemonCard;
