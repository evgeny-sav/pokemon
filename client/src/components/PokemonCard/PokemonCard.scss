.PokemonCard {
  display: flex;
  flex-direction: column;
  width: 300px;
  margin: 15px;
  padding: 20px;

  border: 1px solid #eee;
  border-radius: 5px;

  box-sizing: border-box;
  box-shadow: #ddd 2px 2px 8px;

  .image {
    display: flex;
    align-items: center;
    justify-content: center;
  }

  .title {
    font-weight: bold;
    margin: 10px 0;
  }

  .description {

  }

  .button {
    display: flex;
    flex-flow: row-reverse;

    a {
      display: block;
      width: 100%;
      text-decoration: none;

      button {
        width: 100%;
      }
    }
  }
}
