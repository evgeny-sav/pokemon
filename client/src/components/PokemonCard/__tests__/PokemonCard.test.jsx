import React from 'react';
import { create } from 'react-test-renderer';
import PokemonCard from '../PokemonCard';

describe('PokemonCard', () => {
  it('should render', () => {
    const component = create(
      <PokemonCard pokemon={{ name: 'test', sprites: {} }} />,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
