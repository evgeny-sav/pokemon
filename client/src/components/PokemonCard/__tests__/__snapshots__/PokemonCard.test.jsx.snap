// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCard should render 1`] = `
<div
  className="PokemonCard"
>
  <div
    className="image"
  >
    <img
      alt="Pokemon's image"
    />
  </div>
  <div
    className="cord-body"
  >
    <h5
      className="title"
    >
      Test
    </h5>
    <p
      className="description"
    >
      Some quick example text to build on the card title and make up the bulk of the cards content.
    </p>
    <div
      className="button"
    >
      <Link
        to="/pokemons/undefined"
      >
        <button
          className="btn btn-outline-info"
          disabled={false}
          type="button"
        >
          Open
        </button>
      </Link>
    </div>
  </div>
</div>
`;
